### DISCUSSION TITLE
How to complete non-invasive trace monitoring

### DISCUSSION THREAD

**Discussion by hxd123456** [apache/skywalking#11091] [Q&A] [answered]
I use skywalking to monitor application services, but many parts of application services are concurrent, and various concurrency api are used, such as Thread pool, spring @ Async, parallelStream and other mechanisms. These skywalking have been solved, but for 
 `new Thread(new Runnable() {
	@Override
	public void run() {
		log.info("hello word"+TraceContext.traceId());
	}
}).start();`
I need a lot of time to solve it,because we have a lot of services.
change the code like this
`new Thread(RunnableWrapper.of(new Runnable() {
	@Override
	public void run() {
		log.info("hello word"+TraceContext.traceId());
	}
})).start();`
I want to know what more elegant way to solve it

**Comment by lujiajing1126** [comment]
https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/bootstrap-plugins/#bootstrap-class-plugins

Bootstrap plugins may help for your cases

**Reply by hxd123456** [reply]
I read the Bootstrap plugins ,but the plugins only can resolve thead pool,forkjoin and custom runnable/callable,
for  `new Thread(new Runnable() { @Override public void run() { log.info("hello word"+TraceContext.traceId()); } }).start()` 
can not resolve ,I must to rebuild my code

**Reply by wu-sheng** [reply, marked as answer]
We can't do that, choose toolkit and change source codes.
Manipulating all threads is dangerous.
